src/community/wmts-multi-dimensional/src/main/java/org/geoserver/gwc/wmts/dimensions/DimensionsUtils.java

```diff
@@ -226,6 +236,25 @@ private static Object getMaxValue(List<Object> values) {
         return values;
     }
 
+    /**
+     * Helper method that extracts a set of aggregates on the given collection and attribute and returns the results
+     */
+    static Map<Aggregate, Object> getAggregates(String attributeName, FeatureCollection featureCollection, Aggregate... aggregates) {
+        Map<Aggregate, Object> result = new HashMap<>();
+        PropertyName property = FF.property(attributeName);
+        for (Aggregate aggregate : aggregates) {
+            FeatureCalc featureCalc = aggregate.create(property);
+            try {
+                featureCollection.accepts(featureCalc, null);
+                Object value = featureCalc.getResult().getValue();
+                result.put(aggregate, value);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
```

Comment:
I would have a put a message saying which aggregation failed, it may help debug a failure.

Reply:
Good idea